src/prog_models/utils/calc_error.py

```diff
@@ -6,12 +6,13 @@
 """
 
 from collections.abc import Iterable
+from typing import List
 from warnings import warn
 import math
 import numpy as np
 
 
-def MAX_E(m, times, inputs, outputs, **kwargs):
+def MAX_E(m, times: List[float], inputs: List[dict], outputs: List[dict], **kwargs) -> float:
```

Comment:
```suggestion
def MAX_E(m, times: List[float], inputs: List[InputContainer], outputs: List[OutputContainer], **kwargs) -> float:
```
Is this consistent with our new notation? If so, let's add it here and for the other methods below